File: python/hash-table/longest-palindrome.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: int
        """
        d = {}
        ret = 0
        plus = 0
        for c in s:
            d[c] = d.get(c,0) + 1

        for (key,count) in d.items():
            ret += count
            if count % 2 :
                ret -= 1
                plus = 1

        return ret + plus

File: c/hash-table/longest-palindrome.c
int longestPalindrome(char* s) {
    int map[256] = {0};
    int i,len,ret = 0;
    int plus = 0;

    len = strlen(s);
    for(i = 0; i < len; ++i)
        map[s[i]]++;

    for(i = 0; i < 256; ++i){
        ret += map[i];
        if(map[i] % 2 == 1){
            plus = 1;
            ret--;
        }
    }

    return ret + plus;
}
